tests/load/test_read_interfaces.py

```diff
@@ -212,6 +213,73 @@ def double_items():
     loads_table = pipeline._dataset()[pipeline.default_schema.loads_table_name]
     loads_table.fetchall()
 
+    destination_for_dataset: TDestinationReferenceArg = (
+        alternate_access_pipeline.destination
+        if alternate_access_pipeline
+        else destination_config.destination_type
+    )
+
+    # check dataset factory
+    dataset = dlt.dataset(destination=destination_for_dataset, dataset_name=pipeline.dataset_name)
+    # verfiy that sql client and schema are lazy loaded
+    assert not dataset._schema
+    assert not dataset._sql_client
+    table_relationship = dataset.items
+    table = table_relationship.fetchall()
+    assert len(table) == total_records
+
+    # check that schema is loaded by name
+    dataset = dlt.dataset(
+        destination=destination_for_dataset,
+        dataset_name=pipeline.dataset_name,
+        schema=pipeline.default_schema_name,
+    )
+    assert dataset.schema.tables["items"]["write_disposition"] == "replace"  # type: ignore
+
+    # check that schema is not loaded when wrong name given
+    dataset = dlt.dataset(
+        destination=destination_for_dataset,
+        dataset_name=pipeline.dataset_name,
+        schema="wrong_schema_name",
```

Comment:
we allow a new schema to be added right? that's why we do not raise when schema is not known? we'll need a better method of adding schemas to a dataset. also to sync schemas etc.

Reply:
What actually happens here is that an empty schema is created as a standin, it does not do anything and also does not get saved anywhere. I can change that if you like, but afaik for now it should be ok.

Reply:
no, that is OK. I just thing that all the code that interacts with destination and now is in the pipeline (ie. schema lists, schema storage etc. probably could go to Dataset at some point)